internal/db/schema/migrations/oss/postgres/59/01_ldap.up.sql

```diff
@@ -0,0 +1,354 @@
+  scope_id wt_scope_id not null,
+  name wt_name,
+  description wt_description,
+  create_time wt_timestamp,
+  update_time wt_timestamp,
+  version wt_version,
+  state text not null
+    constraint auth_ldap_method_state_enm_fkey
+      references auth_ldap_method_state_enm(name)
+      on delete restrict
+      on update cascade,
+  start_tls bool not null default false,
+  insecure_tls bool not null default false,
+  discover_dn bool not null default false,
+  anon_group_search bool not null default false,
+  upn_domain text
+    constraint upn_domain_too_short
+      check (length(trim(upn_domain)) > 0)
+    constraint upn_domain_too_long
+      check (length(trim(upn_domain)) < 253),
+  constraint auth_ldap_method_fkey
+    foreign key (scope_id, public_id)
+    references auth_ldap_method (scope_id, public_id)
+    on delete cascade
+    on update cascade,
+  constraint auth_ldap_method_scope_id_name_uq
+    unique(scope_id, name),
+  constraint auth_ldap_method_scope_id_public_id_uq
+    unique(scope_id, public_id)
+);
+comment on table auth_ldap_method is
+'auth_ldap_method entries are the current ldap auth methods configured for '
+'existing scopes';
+
+-- auth_ldap_method column triggers
+create trigger insert_auth_method_subtype before insert on auth_ldap_method
+  for each row execute procedure insert_auth_method_subtype();
+
+create trigger update_time_column before update on auth_ldap_method
+  for each row execute procedure update_time_column();
+
+create trigger immutable_columns before update on auth_ldap_method
+  for each row execute procedure immutable_columns('public_id', 'scope_id', 'create_time');
+
+create trigger default_create_time_column before insert on auth_ldap_method
+  for each row execute procedure default_create_time();
+
+create trigger update_version_column after update on auth_ldap_method
+  for each row execute procedure update_version_column();
+
+-- auth_ldap_url entries are LDAP URLs that specify an LDAP servers to connect
+-- to. Examples: ldap://ldap.myorg.com, ldaps://ldap.myorg.com:636. There must
+-- be at least one and if there's more than one URL configured for an auth
+-- method, the directories will be tried in connection_priority order if there
+-- are errors during the connection process.  The URL scheme must be either ldap
+-- or ldaps. The port is optional.If no port is specified, then a default of 389
+-- is used for ldap and a default of 689 is used for ldaps. (see rfc4516 for
+-- more information about LDAP URLs)  
+--
+-- Updates will be implemented as a delete + insert with the auth_ldap_method
+-- being used as the root aggregate for auth_ldap_url updates. 
+create table auth_ldap_url ( 
+  create_time wt_timestamp,
+  ldap_method_id wt_public_id not null  
+    constraint auth_ldap_method_fkey
+      references auth_ldap_method(public_id)
+      on delete cascade
+      on update cascade,
+  url wt_url not null,
+  connection_priority int not null
```

Comment:
Should there be a unique constraint on the connection priority? Or maybe it should be a part of the primary key?

Reply:
ty!